app/src/components/dialog/RequestDialog.tsx

```diff
@@ -0,0 +1,111 @@
+import Box from '@material-ui/core/Box';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { mdiCheck, mdiWindowClose } from '@mdi/js';
+import Icon from '@mdi/react';
+import { Formik } from 'formik';
+import React from 'react';
+
+export interface IRequestDialogComponent {
+  element: any;
+  initialValues: any;
+  validationSchema: any;
+}
+
+export interface IRequestDialog {
+  /**
+   * The Formik compatible component to render in the dialog content body.
+   *
+   * @type {IRequestDialogComponent}
+   * @memberof IRequestDialog
+   */
+  component: IRequestDialogComponent;
+  /**
+   * The dialog window title text.
+   *
+   * @type {string}
+   * @memberof IRequestDialog
+   */
+  dialogTitle: string;
+  /**
+   * Set to `true` to open the dialog, `false` to close the dialog.
+   *
+   * @type {boolean}
+   * @memberof IRequestDialog
+   */
+  open: boolean;
+  /**
+   * Callback fired if the dialog is closed.
+   *
+   * @memberof IRequestDialog
+   */
+  onClose: () => void;
+  /**
+   * Callback fired if the 'Deny' button is clicked.
+   *
+   * @memberof IRequestDialog
+   */
+  onDeny: () => void;
+  /**
+   * Callback fired if the 'Approve' button is clicked.
+   *
+   * @memberof IRequestDialog
+   */
+  onApprove: (values: any) => void;
+}
+
+const RequestDialog: React.FC<IRequestDialog> = (props) => {
```

Comment:
this file needs tests (not necessarily in this pr)

Reply:
Started on it, but will save for a separate PR